lib/std/crypto/aes/soft.zig

```diff
@@ -410,333 +411,99 @@ pub const Aes256 = struct {
-    0x93ba0ae2, 0xa02ae5c0, 0x22e0433c, 0x1b171d12, 0x090d0b0e, 0x8bc7adf2, 0xb6a8b92d, 0x1ea9c814,
-    0xf1198557, 0x75074caf, 0x99ddbbee, 0x7f60fda3, 0x01269ff7, 0x72f5bc5c, 0x663bc544, 0xfb7e345b,
-    0x4329768b, 0x23c6dccb, 0xedfc68b6, 0xe4f163b8, 0x31dccad7, 0x63851042, 0x97224013, 0xc6112084,
-    0x4a247d85, 0xbb3df8d2, 0xf93211ae, 0x29a16dc7, 0x9e2f4b1d, 0xb230f3dc, 0x8652ec0d, 0xc1e3d077,
-    0xb3166c2b, 0x70b999a9, 0x9448fa11, 0xe9642247, 0xfc8cc4a8, 0xf03f1aa0, 0x7d2cd856, 0x3390ef22,
-    0x494ec787, 0x38d1c1d9, 0xcaa2fe8c, 0xd40b3698, 0xf581cfa6, 0x7ade28a5, 0xb78e26da, 0xadbfa43f,
-    0x3a9de42c, 0x78920d50, 0x5fcc9b6a, 0x7e466254, 0x8d13c2f6, 0xd8b8e890, 0x39f75e2e, 0xc3aff582,
-    0x5d80be9f, 0xd0937c69, 0xd52da96f, 0x2512b3cf, 0xac993bc8, 0x187da710, 0x9c636ee8, 0x3bbb7bdb,
-    0x267809cd, 0x5918f46e, 0x9ab701ec, 0x4f9aa883, 0x956e65e6, 0xffe67eaa, 0xbccf0821, 0x15e8e6ef,
-    0xe79bd9ba, 0x6f36ce4a, 0x9f09d4ea, 0xb07cd629, 0xa4b2af31, 0x3f23312a, 0xa59430c6, 0xa266c035,
-    0x4ebc3774, 0x82caa6fc, 0x90d0b0e0, 0xa7d81533, 0x04984af1, 0xecdaf741, 0xcd500e7f, 0x91f62f17,
-    0x4dd68d76, 0xefb04d43, 0xaa4d54cc, 0x9604dfe4, 0xd1b5e39e, 0x6a881b4c, 0x2c1fb8c1, 0x65517f46,
-    0x5eea049d, 0x8c355d01, 0x877473fa, 0x0b412efb, 0x671d5ab3, 0xdbd25292, 0x105633e9, 0xd647136d,
-    0xd7618c9a, 0xa10c7a37, 0xf8148e59, 0x133c89eb, 0xa927eece, 0x61c935b7, 0x1ce5ede1, 0x47b13c7a,
-    0xd2df599c, 0xf2733f55, 0x14ce7918, 0xc737bf73, 0xf7cdea53, 0xfdaa5b5f, 0x3d6f14df, 0x44db8678,
-    0xaff381ca, 0x68c43eb9, 0x24342c38, 0xa3405fc2, 0x1dc37216, 0xe2250cbc, 0x3c498b28, 0x0d9541ff,
-    0xa8017139, 0x0cb3de08, 0xb4e49cd8, 0x56c19064, 0xcb84617b, 0x32b670d5, 0x6c5c7448, 0xb85742d0,
+    break :init array;
 };
+
+const sbox_encrypt align(64) = generateSbox(false);
+const sbox_decrypt align(64) = generateSbox(true);
+const table_encrypt align(64) = generateTable(sbox_encrypt, [_]u8{ 0x3, 0x1, 0x1, 0x2 });
+const table_decrypt align(64) = generateTable(sbox_decrypt, [_]u8{ 0xb, 0xd, 0x9, 0xe });
+
+// Generate S-box substitution values.
+fn generateSbox(invert: bool) [256]u8 {
+    var array: [256]u8 align(64) = undefined;
+
+    var p: u8 = 1;
+    var q: u8 = 1;
+    for (array) |_| {
+        p = mul(p, 3);
+        q = mul(q, 0xf6); // divide by 3
+
+        var value: u8 = q ^ 0x63;
+        value ^= math.rotl(u8, q, 1);
+        value ^= math.rotl(u8, q, 2);
+        value ^= math.rotl(u8, q, 3);
+        value ^= math.rotl(u8, q, 4);
+
+        if (invert) {
+            array[value] = p;
+        } else {
+            array[p] = value;
+        }
+    }
+
+    if (invert) {
+        array[0x63] = 0x00;
+    } else {
+        array[0x00] = 0x63;
+    }
+
+    return array;
+}
+
+// Generate lookup tables for encryption and decryption.
+fn generateTable(sbox: [256]u8, factors: [4]u8) [4][256]u32 {
+    var array: [4][256]u32 align(64) = undefined;
```

Comment:
Do we want/need the `align(64)` here?
Does return value semantics kick in without it?